### PR TITLE
Add domainEnrichment function to ESA

### PR DESCRIPTION
**Pull request by VVVVVan**
Add domainEnrichment function and its vignette.

### CODE REVIEW

**Comment by hyginn** [comment]
I resolved your NAMESPACE conflict so Travis could begin to check. Waiting ...

**Comment by VVVVVan** [comment]
Hi Prof Steipe,

I feel a little bit confused. My check has passed long time ago but when someone approves and merges one's pull request, I will have conflicts in description and namespace. Is that mean that whenever someone merges pull request, I need to resolve the conflicts? or I just leave it there?

Thanks,
Fan

File: R/domainEnrichment.R [hunk 1]
```diff
@@ -0,0 +1,164 @@
+# domainEnrichment.R
+
+#' Domain enrichment in a desired system.
+#'
+#' \code{domainEnrichment} Find the enriched domains in selected system and interpret enrichment by plot.
+#'
+#' For the PHYLA system, see details in the \href{../doc/domainEnrichment-Vignette.html}{PHALY domain enrichemnt vignette} (or load the vignette with \code{vignette("domainEnrichment-Vignette", package = "BCB420.2019.ESA")}).
```
Comment (hyginn) [inline R/domainEnrichment.R:7]: Lines should be <80 characters ... BiocCheck will complain.

File: R/domainEnrichment.R [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+#'
+#' @section Details: This function is created as part of BCB420 2019 winter course. The domain databaase were fetched from InterPro, see more details in \code{\link{fetchData}}.
+#'
+#' @param system The shortehand notation of selected system, five-letter code as discussed in class.
```
Comment (hyginn) [inline R/domainEnrichment.R:11]: Spelling, spelling, spelling. And grammar.

File: R/domainEnrichment.R [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+#'
+#' @examples
+#' # Plot enrichend domains in PHALY system orderd by rich factor
+#' # Plot would separate to two parts if too many domains are enriched
+#' # User can hover around the point on plot to get more info (i.e ID descripton)
+#' system <- "PHALY"
+#' domainEnrichment(system)
+#'
+#' @export
+domainEnrichment <- function(system, alpha = 0.05) {
```
Comment (hyginn) [inline R/domainEnrichment.R:34]: Good on you to allow the user to set alpha.

File: R/domainEnrichment.R [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  # 1. Fetch data from Dr. Steipe's BCB420.2019.ESA package
+  # Two lists from InterPro database
```
Comment (hyginn) [inline R/domainEnrichment.R:36]: Fetching data inside the function is not a good idea. Better to pass it as a parameter, which can default to fetching it, if it doesn't already exist in the workspace.

File: R/domainEnrichment.R [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  genesIPR <- fetchData("genesIPR")
+  IPRgenes <- fetchData("IPRgenes")
+
+  # Genes for the system
+  myDB <- fetchData("SysDB")
+  if (! system %in% names(SyDBgetRootSysIDs(myDB))) {
```
Comment (hyginn) [inline R/domainEnrichment.R:42]: Same ...

File: R/domainEnrichment.R [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+  # 3.4 Multiple testing: Benjamini-Hochberg control for false discovery rate
+  tmp$p_adj_BH <- p.adjust(tmp$p_value, "BH")
+
+  # 3.5 Calculate the enrichment
+  tmp$enrichment <- (tmp$sysFreq)/(tmp$randomFreq)
+
+
+  # 4. Find the description for enriched domains
+  myMart <- biomaRt::useMart("ensembl", dataset="hsapiens_gene_ensembl")
```
Comment (hyginn) [inline R/domainEnrichment.R:104]: That will make the function unbearably slow. I would precompile the domain descriptions. ToDo ...

File: R/domainEnrichment.R [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+  description <- biomaRt::getBM(filters = "interpro",
+                                attributes = c("interpro",
+                                               "interpro_short_description",
+                                               "interpro_description"),
+                                values = tmp$domains,
+                                mart = myMart)
+
+  # Merge the two data frame
+  # note: domain IDs with no description are excluded since they are family
```
Comment (hyginn) [inline R/domainEnrichment.R:114]: ??? Not sure what this means.

**Review by hyginn — APPROVED** [review]
Looking good overall.

**Comment by hyginn** [comment]
Good. Merged.

**Pull request merged**